internal/users/idgenerator.go

```diff
@@ -0,0 +1,265 @@
+package users
```

Comment:
Note to self: Write a commit message for the ID generation

Reply:
```
Pick deterministic UIDs/GIDs instead of generating random ones

Now that we have a way to effectively avoid races when picking a UID/GID
(assuming that all other system tools and services also use lckpwdf,
which they should), there is no good reason anymore to pick random IDs
from such a large ID range. So with this commit we pick deterministic
IDs, starting with the next higher ID above the highest ID already in
use by an authd user or a user in `/etc/passwd` (while still respecting
the configured UID_MIN value).

In follow-up work, this will also allow us to change the default UID/GID
ranges to a smaller range which doesn't overlap with the LXD ID ranges,
which should fix #819.
```